crux-test/test/crux/fixtures/instrument.clj

```diff
@@ -4,99 +4,180 @@
             [crux.memory :as mem]
             [clojure.tools.logging :as log]))
 
-(defn- i-name [i]
-  ;;(-> i ^java.lang.Class type .getName (clojure.string/replace #"crux\.index\." "") symbol)
-  (-> i str (clojure.string/replace #"crux\.index\." "")))
+(def ansi-colors {:white   "[37m"
+                  :red     "[31m"
+                  :green   "[32m"
+                  :blue    "[34m"
+                  :yellow  "[33m"
+                  :magenta "[35m"})
+
+(defn- in-ansi [c s]
+  (if c
+    (str "\u001b" (get ansi-colors c) s "\u001b[0m")
+    s))
+
+(defmulti index-name (fn [i] (-> i ^java.lang.Class type .getName symbol)))
+
+(defmethod index-name :default [i]
+  (-> i ^java.lang.Class type .getName (clojure.string/replace #"crux\.index\." "") symbol))
+
+(defmethod index-name 'crux.index.NAryConstrainingLayeredVirtualIndex [i]
+  (str "NAry-Constrained: " (clojure.string/join " " (map :name (:indexes i)))))
+
+(defmethod index-name 'crux.index.NAryJoinLayeredVirtualIndex [i]
+  (str "NAry: " (clojure.string/join " " (map :name (:indexes i)))))
+
+(defmethod index-name 'crux.index.UnaryJoinVirtualIndex [i]
+  (str "Unary: " (clojure.string/join " " (map :name (:indexes i)))))
+
+(defmethod index-name 'crux.index.BinaryJoinLayeredVirtualIndex [i]
+  (format "Binary: [%s %s %s]" (-> i meta :clause :e) (-> i meta :clause :a) (-> i meta :clause :v)))
+
+(defmethod index-name 'crux.index.DocAttributeValueEntityEntityIndex [i]
+  "AVE-E:")
+
+(defmethod index-name 'crux.index.DocAttributeValueEntityValueIndex [i]
+  "AVE-V:")
+
+(defmethod index-name 'crux.index.DocAttributeEntityValueEntityIndex [i]
+  "AVE-E:")
+
+(defmethod index-name 'crux.index.DocAttributeEntityValueValueIndex [i]
+  "AEV-V:")
 
 (defn- trunc
   [s n]
   (subs s 0 (min (count s) n)))
 
-(defn- trace-op [foo op depth & extra]
-  (println (format "%s %s%s %s" (name op) @foo (apply str (take depth (repeat " ")))
-                   (clojure.string/join " " extra))))
+(defn- trace-op [{:keys [foo depth color] :as this} op & extra]
```

Comment:
newbie question (don't worry about changing the PR) - what's `foo`? (aware it was already in instrument.clj before the PR)

Reply:
Apologies - changes to `level`